.github/workflows/test-linux.yml

```diff
@@ -0,0 +1,125 @@
+---
+name: Test Linux
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: >
+          Python version to currently test
+        type: string
+        required: true
+      runner:
+        description: >
+          Describes the system this workflow should run on.
+        type: string
+        required: true
+      use-msrv:
+        description: >
+          Decides whether we use MSRV or stable for rust toolchain.
+        required: false
+        type: boolean
+        default: false
+      install-optionals:
+        description: >
+          Decides whether we install optional dependencies
+        type: boolean
+        default: false
+      install-from-sdist:
+        description: >
+          Decides whether we perform rust tests
+        type: boolean
+        default: false
+
+jobs:
+  unit-tests:
+    if: github.repository_owner == 'Qiskit'
+    name: ${{ inputs.runner }}-tests-Python-${{ inputs.python-version }}
+    runs-on: ${{ inputs.runner }}
+    timeout-minutes: 60
+    env:
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@1.79
+        if: ${{ inputs.use-msrv }}
```

Comment:
This is actually just likely to get out of sync, if `use-msrv` is set to true you're potentially installing rust twice if install from sdist is used and not actually running with a different rust versions. You would need to also set the appropriate override when calling cargo (via pip) if you wanted `false` case to be respected fully.

I would just remove this flag and option from the test jobs. This is different from the azure config as that always used MSRV and never ran with Rust stable (it did over a year ago but we moved away from that). We can discuss doing this in the future, but we shouldn't make changes like this as part of a migration PR.

Reply:
Removed in 8d54c24